### PR TITLE
Refactor Eventing Components

### LINKED ISSUE #3024: Consistent naming for eventing production components
>Note:
The names need to be discussed with @a-thaler 

This might involve changes in

* publish-knative
* subscription-controller-knative

Refer to driving topic https://github.com/kyma-project/kyma/issues/2502

**AC**
- [ ] The eventing production components have been renamed consistently.

### PR DESCRIPTION
**Pull request by anishj0shi**
Description
As per #2502 certain components from the event-bus have to renamed and this PR proposes changes to comply that.

Changes proposed in this pull request:

Refactor event-bus-subscription-controller-knative => subscription-controller
Refactor event-bus-publish-knative => event-publish-service
Related issue(s)

Resolves #3024

### CODE REVIEW

File: components/event-bus/internal/knative/util/kn-eventing.go [hunk 1]
```diff
@@ -201,19 +192,16 @@ func (k *KnativeLib) UpdateSubscription(sub *evapisv1alpha1.Subscription) (*evap
 
 // SendMessage sends a message to a channel
 func (k *KnativeLib) SendMessage(channel *evapisv1alpha1.Channel, headers *map[string][]string, payload *string) error {
-
+	httpClient := &http.Client{
+		Transport: initHTTPTransport(),
+	}
```
Comment (antoineco) [inline components/event-bus/internal/knative/util/kn-eventing.go]: What was the motivation behind the replacement of that singleton?

Comment (anishj0shi) [reply components/event-bus/internal/knative/util/kn-eventing.go]: commit b59a48a56125502abd0a71ddae99008d10e65164 was missing on this PR, this issue has been addressed by Rafaey.

File: components/event-bus/README.md [hunk 2]
```diff
@@ -8,14 +8,14 @@ The Event Bus enables Kyma to integrate with various other external solutions. T
 
 Currently, Event Bus makes the following three Docker images available to the `kyma core` Helm chart:
```
[the remark this answers is not in the document]
Comment (anishj0shi) [reply components/event-bus/README.md]: done

File: components/event-bus/README.md [hunk 3]
```diff
@@ -8,14 +8,14 @@ The Event Bus enables Kyma to integrate with various other external solutions. T
...
 
-- event-bus-publish-knative
-- event-bus-subscription-controller-knative
+- event-publish-service
+- subscription-controller
```
[the remark this answers is not in the document]
Comment (anishj0shi) [reply components/event-bus/README.md]: done

File: components/event-bus/README.md [hunk 4]
```diff
@@ -8,14 +8,14 @@ The Event Bus enables Kyma to integrate with various other external solutions. T
...
 
 There are also end-to-end test Docker images to use as `helm tests`. See [the tests in the `event-bus` directory](https://github.com/kyma-project/kyma/tree/master/tests/event-bus) for more details.
```
[the remark this answers is not in the document]
Comment (anishj0shi) [reply components/event-bus/README.md]: done

**Pull request merged**